### DISCUSSION TITLE
useRowSelect hook and infinite loop

### DISCUSSION THREAD

**Discussion by r-s-sandbox** [TanStack/table#2905] [Q&A]
Hi All,
I would appreaciate some help with a table selection issue and the callback not firing in the codesandbox (https://codesandbox.io/s/zealous-paper-v8xrg). If I remove the useCallback hook, it sends the component into an infinite loop.

**Comment by saurabhgupta11** [comment]
I was facing the same issue, using selectedFlatRows as a dependency in useEffect causes an infinite loop
This link has an alternate solution, by re-implementing selectedFlatRows using useMountedLayoutEffect and using selectedRowIds as a dependency in useEffect - [(https://github.com/tannerlinsley/react-table/issues/1739)]

**Reply by handhikadj** [reply]
> I was facing the same issue, using selectedFlatRows as a dependency in useEffect causes an infinite loop This link has an alternate solution, by re-implementing selectedFlatRows using useMountedLayoutEffect and using selectedRowIds as a dependency in useEffect - [(https://github.com/[/issues/1739](https://github.com/TanStack/react-table/issues/1739))]

would be helpful if you can provide codesandbox on this
